src/rest.js

```diff
@@ -50,7 +61,11 @@ export class Rest {
         return response.json().catch(error => null);
       }
 
-      throw response;
+      if (parseError) {
+        return response.json().catch(() => response).then(error => Promise.reject(error));
```

Comment:
the response might not be a json. hence catching it like aver in line 61. so error will be either the response or JSON.parse(response) there. 

i used Promise.reject due to an error in thinking. what i meant is than one should throw resp. reject with type Error. that's now tricky to do BC unless we add an option for that